docs/CreatingSH.md

```diff
@@ -0,0 +1,35 @@
+
+After the configuration, the core threads each system handle, calling its “spin_once” function repeatedly. The communication between different systems is thus managed as follows:
+
+![](http://www.plantuml.com/plantuml/png/VOyn2uCm48Nt_8h3oGuEPYobStVNWbpFO88n2IP5-kzxY1A7u9XtttiV4G6NPCW4T0cgxXUJcjzEenkiWhO2ZD2zsajAxfGiKST6SUAeIY76nNy3hDhme9_gcs0hD4ykmXr8Avhwv0EP2BKFoNY7bfaDOP8PfzP-ZcFauYcD2XVIFQ6r7wJfT9LyPPu3kGN7EmTNOfb7JkASbiMF9elp1_UtSiCV)
+
+# Developing a new system handle
+
+When developing a new system handle, a class must be implemented with some specific functions. The number of functions needed vary with the type of system implemented, for example [FIWARE system handle][FW_SH] will not need to use the server/client part of soss, so it will only have the functions to publish and advertise, among some common functions that must always be implemented. In that case, just a class inheriting from `soss::TopicSystem` will hold all the functions and classes needed.
+
+The first function to be implemented is `configure`, which receives from the core a list with the names of the required types that will be used in each run. The system handle can use that list to load the libraries needed to convert between soss messages and system messages, if it is necessary.
+
+There is also a function called `okay` with which the core must be able to check if the system handle is still running properly. Another function, called `spin_once`, will have the purpose of managing the communications with the external system. For example, in ROS2 system handle, that function calls an analogue function in ROS2 to make it check if there is any new message. This function will be called repeatedly from the soss core.
+
+The function advertise will receive a topic name and a message type, and it will have to retrieve a class called `TopicPublisher`, that must also be implemented, and that will have a function called `publish`, which will receive a soss message and transform it to a type of message that the involved system can understand, sending it to the system.
+
+Last, there must be a function `subscribe` that will receive the topic name and type to which it has to subscribe, and a callback function, which will be one of the `publish` functions handled to the core from another system handle. That `subscribe` function must first create a subscription in its system and then store the callback in a way that it will be invoked each time a new message arrives to the corresponding topic, converting it to a soss message first.
```

Comment:
I would change
`which will be one of the publish functions handled to the core from another system handle`
to
`which will be called from the publisher of another system handler`